xbmc/utils/Signals.h

```diff
@@ -0,0 +1,201 @@
+/*
```

Comment:
> If you have concrete ideas on how the "future" should look like we can try them out here though :-)

Have a look at ActiveAE. The concept is based on UML ports and protocols. The state machine makes it easy to separate aspects of behaviour.